lib/src/signed_video_h26x_auth.c

```diff
@@ -114,6 +114,24 @@ decode_sei_data(signed_video_t *self, const uint8_t *payload, size_t payload_siz
   return status;
 }
 
+/**
+ * Detects if there are any missing SEI messages based on the GOP counter and updates the GOP state.
+ */
+static void
+detect_sei_loss_and_sync_gop(signed_video_t *self)
+{
+  // Get the last GOP counter before updating.
+  uint32_t last_gop_number = self->gop_info->latest_validated_gop;
+  uint32_t exp_gop_number = last_gop_number + 1;
```

Comment:
Looks like a copy-paste thing, but why do you need `last_gop_number`? I think it is better to compute `exp_gop_number` at once.